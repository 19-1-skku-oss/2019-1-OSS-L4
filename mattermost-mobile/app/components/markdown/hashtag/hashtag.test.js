// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {shallow} from 'enzyme';
import React from 'react';
import {Text} from 'react-native';

import Hashtag from './hashtag';

describe('Hashtag', () => {
    const baseProps = {
        hashtag: 'test',
        linkStyle: {color: 'red'},
        navigator: {
            dismissAllModals: jest.fn(),
            popToRoot: jest.fn(),
        },
        actions: {
            showSearchModal: jest.fn(),
        },
    };

    test('should match snapshot', () => {
        const wrapper = shallow(<Hashtag {...baseProps}/>);

        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should open hashtag search on click', () => {
        const props = {
            ...baseProps,
        };

        const wrapper = shallow(<Hashtag {...props}/>);

        wrapper.find(Text).simulate('press');

        expect(props.navigator.dismissAllModals).toHaveBeenCalled();
        expect(props.navigator.popToRoot).toHaveBeenCalled();
        expect(props.actions.showSearchModal).toHaveBeenCalledWith(props.navigator, '#test');
    });

    test('should call onHashtagPress if provided', () => {
        const props = {
            ...baseProps,
            onHashtagPress: jest.fn(),
        };

        const wrapper = shallow(<Hashtag {...props}/>);

        wrapper.find(Text).simulate('press');

        expect(props.navigator.dismissAllModals).not.toBeCalled();
        expect(props.navigator.popToRoot).not.toBeCalled();
        expect(props.actions.showSearchModal).not.toBeCalled();

        expect(props.onHashtagPress).toBeCalled();
    });
});
