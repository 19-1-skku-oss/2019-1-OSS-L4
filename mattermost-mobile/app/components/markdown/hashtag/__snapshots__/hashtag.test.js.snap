// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hashtag should match snapshot 1`] = `
<Text
  onPress={[Function]}
  style={
    Object {
      "color": "red",
    }
  }
>
  #test
</Text>
`;
